TodoApi/Extensions/SerilogExtensions.cs

```diff
@@ -0,0 +1,132 @@
+
+            loggerConfiguration
+                .Enrich.WithProperty("Application", "TodoApi")
+                .Enrich.FromLogContext()
+                .Enrich.WithEnvironmentName()
+                .Enrich.WithMachineName()
+                // https://rehansaeed.com/logging-with-serilog-exceptions/
+                .Enrich.WithExceptionDetails();
+
+            if (context.HostingEnvironment.IsDevelopment())
+            {
+                // https://github.com/serilog/serilog-sinks-async
+                loggerConfiguration.WriteTo.Async(writeTo => writeTo.Console(
+                    logLevel,
+                    loggerOptions.LogTemplate
+                ));
+            }
+
+            if (!string.IsNullOrEmpty(loggerOptions.ElasticSearchUrl))
+            {
+                // https://github.com/serilog-contrib/serilog-sinks-elasticsearch
+                loggerConfiguration.WriteTo.Async(
+                    writeTo =>
+                        writeTo.Elasticsearch(ConfigureElasticSink(
+                            loggerOptions.ElasticSearchUrl,
+                            context.HostingEnvironment.EnvironmentName)));
+            }
+
+            if (!string.IsNullOrEmpty(loggerOptions.SeqUrl))
+            {
+                loggerConfiguration.WriteTo.Async(writeTo => writeTo.Seq(loggerOptions.SeqUrl));
+            }
+
+            if (!string.IsNullOrEmpty(loggerOptions.LogPath))
+            {
+                loggerConfiguration.WriteTo.File(
+                    loggerOptions.LogPath,
+                    outputTemplate: loggerOptions.LogTemplate,
+                    rollingInterval: RollingInterval.Day,
+                    rollOnFileSizeLimit: true);
+            }
+        });
+
+        return builder;
+    }
+
+    private static ElasticsearchSinkOptions ConfigureElasticSink(string elasticUrl, string environment)
+    {
+        return new(new Uri(elasticUrl))
+        {
+            AutoRegisterTemplate = true,
+            IndexFormat =
+                $"{Assembly.GetEntryAssembly()?.GetName().Name?.ToLower(CultureInfo.InvariantCulture)
+                    .Replace(".", "-", StringComparison.OrdinalIgnoreCase)}-{environment.ToLower(CultureInfo.InvariantCulture)
+                    .Replace(".", "-", StringComparison.OrdinalIgnoreCase)}-{DateTime.UtcNow:yyyy-MM}"
+        };
+    }
+}
+
+public class SerilogOptions
+{
+    public string? SeqUrl { get; set; } = default!;
+    public string? ElasticSearchUrl { get; set; } = default!;
+    public string LogTemplate { get; set; } =
+        "{Timestamp:yyyy-MM-dd HH:mm:ss} [{Level:u4}] {Message:lj}{NewLine}{Exception} {Properties:j}";
+    public string? LogPath { get; set; } = default!;
+}
+
+// Ref: https://andrewlock.net/using-serilog-aspnetcore-in-asp-net-core-3-logging-the-selected-endpoint-name-with-serilog/
+public static class RequestLogEnricher
+{
+    public static void EnrichFromRequest(IDiagnosticContext diagnosticContext, HttpContext httpContext)
```

Comment:
Doesn't feel like this should be here by default. How expensive is this?

Reply:
In fact, lets just remove it.

Reply:
I remove this enricher